bentoml/repository/__init__.py

```diff
@@ -202,15 +204,58 @@ def dangerously_delete(self, bento_name, bento_version):
 
         try:
             response = self.s3_client.delete_object(Bucket=self.bucket, Key=object_name)
-
             DELETE_MARKER = 'DeleteMarker'  # whether object is successfully deleted.
+
+            # Note: as of boto3 v1.13.13. delete_object returns an incorrect format as
+            # expected from documentation.
+            # Expected format:
+            # {
+            #   'DeleteMarker': True|False,
+            #   'VersionId': 'string',
+            #   'RequestCharged': 'requester'
+            # }
+            # Current return:
+            # {
+            #   'ResponseMetadata': {
+            #     'RequestId': '****************',
+            #     'HostId': '*****/******',
+            #     'HTTPStatusCode': 204,
+            #     'HTTPHeaders': {
+            #       'x-amz-id-2': '*****/xxxxx',
+            #       'x-amz-request-id': '332EE9F7AB555D2B',
+            #        'date': 'Tue, 19 May 2020 19:46:57 GMT',
+            #        'server': 'AmazonS3'
+            #     },
+            #     'RetryAttempts': 0
+            #   }
+            # }
+            # An open issue on github: https://github.com/boto/boto3/issues/759
+            if DELETE_MARKER in response:
+                return response[DELETE_MARKER]
+            elif 'ResponseMetadata' in response:
+                # Note: Use head_object to 'check' is the object deleted or not.
+                # head_object only try to retrieve the metadata without returning
+                # the object itself.
+                try:
+                    self.s3_client.head_object(Bucket=self.bucket, Key=object_name)
+                    raise YataiRepositoryException(
+                        f'Fail to delete {bento_name}:{bento_version} in bucket: '
+                        f'{self.bucket}'
+                    )
+                except ClientError as e:
+                    error_response = e.response.get('Error', {})
+                    error_code = error_response.get('Code', 'Unknown')
+                    if error_code != '404':
+                        raise e
```

Comment:
should it return True here?

Reply:
nvm, I think it should be consistent with LocalRepository implementation, see the other comment

Reply:
Add an inline comment?
```
# 'Error code 404 means target file object does not exist, as expected after delete_object call'
```

Reply:
```
if error_code == '404':
    # 'Error code 404 means target file object does not exist, as expected after delete_object call'
    return
else:
    # unexpected boto3 ClientError
    raise e
```